study-frontend/src/Pages/SearchPage.tsx

```diff
@@ -0,0 +1,75 @@
+
+import ClosedMenu from '../Components/Header';
+import { Menu } from '../Components/Menu';
+import TeacherCard from '../Components/TeacherCard';
+import FetchAllUsers from '../Services/User/FetchAllUsers';
+import React, { useState, useEffect } from "react";
+import '../cssFiles/searchPage.css'
+
+interface User {
+  id : number,
+  username: string,
+  email : string
+  password: string
+  
+  
+}
+
+const SearchPage = () => {
+
+  const [users, setUsers] = useState<User[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [clickedFa, setClickedFa] = useState(false);
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const data = await FetchAllUsers();
+        setUsers(data);
```

Comment:
Move this to app state you will probably need the user in other components and its better to not make another request the get him